pandas/core/indexes/multi.py

```diff
@@ -1821,29 +1830,39 @@ def is_lexsorted(self) -> bool:
         In the below examples, the first level of the MultiIndex is sorted because
         a<b<c, so there is no need to look at the next level.
 
-        >>> pd.MultiIndex.from_arrays([['a', 'b', 'c'], ['d', 'e', 'f']]).is_lexsorted()
+        >>> pd.MultiIndex.from_arrays([['a', 'b'], ['d', 'e']])._is_lexsorted()
         True
-        >>> pd.MultiIndex.from_arrays([['a', 'b', 'c'], ['d', 'f', 'e']]).is_lexsorted()
+        >>> pd.MultiIndex.from_arrays([['a', 'b'], ['d', 'f']])._is_lexsorted()
         True
 
         In case there is a tie, the lexicographical sorting looks
         at the next level of the MultiIndex.
 
-        >>> pd.MultiIndex.from_arrays([[0, 1, 1], ['a', 'b', 'c']]).is_lexsorted()
+        >>> pd.MultiIndex.from_arrays([[0, 1, 1], ['a', 'b', 'c']])._is_lexsorted()
         True
-        >>> pd.MultiIndex.from_arrays([[0, 1, 1], ['a', 'c', 'b']]).is_lexsorted()
+        >>> pd.MultiIndex.from_arrays([[0, 1, 1], ['a', 'c', 'b']])._is_lexsorted()
         False
         >>> pd.MultiIndex.from_arrays([['a', 'a', 'b', 'b'],
-        ...                            ['aa', 'bb', 'aa', 'bb']]).is_lexsorted()
+        ...                            ['aa', 'bb', 'aa', 'bb']])._is_lexsorted()
         True
         >>> pd.MultiIndex.from_arrays([['a', 'a', 'b', 'b'],
-        ...                            ['bb', 'aa', 'aa', 'bb']]).is_lexsorted()
+        ...                            ['bb', 'aa', 'aa', 'bb']])._is_lexsorted()
         False
         """
-        return self.lexsort_depth == self.nlevels
+        return self._get_lexsort_depth == self.nlevels
 
-    @cache_readonly
     def lexsort_depth(self):
+        warnings.warn(
+            "MultiIndex.lexsort_depth is deprecated as a public function, "
+            "users should use MultiIndex.is_monotonic_increasing to check "
+            "is a MultiIndex is sorted.",
+            FutureWarning,
+            stacklevel=2,
+        )
+        return self._get_lexsort_depth()
+
+    @cache_readonly
+    def _get_lexsort_depth(self):
```

Comment:
there's already a `_lexsort_depth` function, so am replacing this with `_get_lexsort_depth`

Reply:
umm that is extra confusing. pls don't do this

Reply:
you can just move the .sortorder stuff to _lexsort_depth

Reply:
Thanks for your review - the difficulty I'm running into is that in `pandas/core/indexes/multi.py` there's
```python
        if self.sortorder is not None:
            if self.sortorder > self._lexsort_depth():
                raise ValueError(
                    "Value for sortorder must be inferior or equal to actual "
                    f"lexsort_depth: sortorder {self.sortorder} "
                    f"with lexsort_depth {self._lexsort_depth()}"
                )
```
so I can't just move `self.sortorder` into `self._lexsort_depth` else this will never raise, as `self._lexsort_depth()` would always just return `self.sortorder`.

For now I've factored out the part which is used in the above check into
```
    def _codes_lexsort_depth(self) -> int:
        int64_codes = [ensure_int64(level_codes) for level_codes in self.codes]
        for k in range(self.nlevels, 0, -1):
            if libalgos.is_lexsorted(int64_codes[:k]):
                return k
        return 0
```
but I'd imagine this will be considered equally confusing. Any suggestions?